src/mol-plugin-ui/viewport.tsx

```diff
@@ -67,9 +72,17 @@ export class ViewportControls extends PluginUIComponent<ViewportControlsProps, V
         this.plugin.helpers.viewportScreenshot?.download();
     };
 
+    enableCameraReset = (enable: boolean) => {
+        this.setState(old => ({ ...old, isCameraResetEnabled: enable }));
+    };
+
     componentDidMount() {
         this.subscribe(this.plugin.events.canvas3d.settingsUpdated, () => this.forceUpdate());
         this.subscribe(this.plugin.layout.events.updated, () => this.forceUpdate());
+        if (this.plugin.canvas3d) {
+            this.subscribe(this.plugin.canvas3d.camera.stateChanged, snapshot => this.enableCameraReset(snapshot.radius !== 0 && snapshot.radiusMax !== 0));
+            // I have no idea what radius and radiusMax mean, but this is the combination that seems to works
+        }
```

Comment:
Why do you need to disable this?

This is very ineffective given how often the `stateChanged` is called when interacting with the canvas. If you really want to keep this in, at least add a `.pipe` with `throtte/debounce`.

Alternatively, you can just make the reset a non-action if this condition is met.

Reply:
I checked this, and found out that `stateChanged` does not fire when using mouse or touchpad to move camera.
There is one exception though: pressing middle button and dragging does trigger `stateChanged`.
Is this the expected behavior?

Reply:
You are right, we actually use similar code in other places as well, e.g., https://github.com/molstar/molstar/blob/master/src/mol-plugin-ui/left-panel.tsx#L145

Reply:
I added `.pipe(throttleTime(...))` to all subscriptions to `stateChanged`.